File: js/quickSort.js
function quickSort(arr) {
	if(arr.length < 2) return arr;
	let pvot = arr[Math.floor(Math.random() * arr.length)];
	let left = [];
	let right = [];
	let equal = [];

	for(let value of arr){
		if(pvot > value)
			left.push(value);
		else if(pvot < value)
			right.push(value);
		else
			equal.push(value);
	}

	return Array().concat(quickSort(left)).concat(equal).concat(quickSort(right));
}

let arr = [4,5,7,9,0, 10, 55, 0,0,0,1];
console.log(quickSort(arr))

File: paython/quickSort.py
from random import randint

def quickSort(arr: list[int]) -> None:
    if len(arr) < 2:
        return arr

    pvot = arr.pop(randint(0, len(arr) - 1))
    left = [i for i in arr if i < pvot]
    right = [i for i in arr if i >= pvot]
    return quickSort(left) + [pvot] + quickSort(right)

arr = [5,3,8,7,2,3,10,0,0,5,7]
print(quickSort(arr))
